### PR TITLE
Added crypto-square test template

### LINKED ISSUE #1939: crypto-square: add test template
## Task
Create a test template file for `crypto-square`

### Details
1. If this is your first time contributing to this repository, welcome!
    - Contributing to a repository on GitHub requires a [fork](https://help.github.com/en/articles/fork-a-repo)
1. Create a branch on your local repository called `crypto-square-add-test-template`
    - Run: `git checkout -b crypto-square-add-test-template`
1. Clone problem-specifications repository; this is used to populate the template
    - Run: `git clone https://github.com/exercism/problem-specifications ../problem-specifications`
1. Create `exercises/crypto-square/.meta/template.j2`
    - See [Generator Documentation] for more information
1. Generate test file from new template
    - Run: `bin/generate_tests.py crypto-square`

At this point, if the generator fails, or you are otherwise stuck,
please feel free to reach out! It is perfectly acceptable to submit
an incomplete PR in order to get feedback or assistance.

### Resources
- [Contributing Guide]
- [Generator Documentation]
- [Closing issues using keywords]

[Contributing Guide]: https://github.com/exercism/python/blob/master/docs/CONTRIBUTING.md
[Generator Documentation]: https://github.com/exercism/python/blob/master/docs/GENERATOR.md
[Closing issues using keywords]: https://help.github.com/en/articles/closing-issues-using-keywords

### PR DESCRIPTION
**Pull request by MrNaif2018**
This pull request adds test template for crypto-square task. 
Closes #1939

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,8 @@ tmp
 bin/configlet
 bin/configlet.exe
 .idea/
+.vscode/
+.mypy_cache/
```
Comment (cmccandless) [inline .gitignore]: Please revert this change. Suggest adding to a [global gitignore](https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore) or your local `.git/info/exclude` file.

File: exercises/crypto-square/crypto_square_test.py [hunk 2]
```diff
@@ -1,36 +1,50 @@
...
-    def test_empty_string(self):
-        self.assertEqual(cipher_text(''), '')
+    def test_empty_plaintext_results_in_an_empty_ciphertext(self):
+        value = ""
+        expected = ""
+        self.assertEqual(ciphertext(value), expected)
 
     def test_lowercase(self):
-        self.assertEqual(cipher_text('A'), 'a')
+        value = ""
```
Comment (cmccandless) [inline exercises/crypto-square/crypto_square_test.py]: Seems your template is checking the incorrect key here.

File: .gitignore (hunk 1 above)
Comment (MrNaif2018) [reply .gitignore]: But vscode is an IDE just like pycharm(.idea folder), and mypy cache comes from built-in mypy. Shoudn't it be included as some people code in vs code and might submit PR's from that too?

Comment (cmccandless) [reply .gitignore]: I had no idea what `.idea/` was used for; that line in the `.gitignore` preceeds my involvement with the project.

General policy is to not include environment-specific content in the repository, but as this is a project that receives a lot of beginner contributors that may not know to not check in such files, I suppose we can make an exception for the gitignore file.

**Comment by MrNaif2018** [comment]
image
It imports wrong function. Should I modify imports in header? Or what should I do?

**Comment by MrNaif2018** [comment]
Actually it looks like in test data it is named ciphertext, and in example it is cipher_text

File: exercises/crypto-square/.meta/template.j2 [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{%- import "generator_macros.j2" as macros with context -%}
+{{ macros.header() }}
```
Comment (cmccandless) [inline exercises/crypto-square/.meta/template.j2]:
```suggestion
{{ macros.header(["cipher_text"]) }}
```

**Comment by MrNaif2018** [comment]
Now the tests run fine

**Comment by cmccandless** [comment]
I'm not sure why the Travis build won't run for this PR. I will investigate tomorrow.

**Comment by MrNaif2018** [comment]
It is because it looks like they somehow flagged me for abuse, going to contact them and figure out the problem. But tests pass I believe (:

**Comment by MrNaif2018** [comment]
To re-trigger the build.

**Comment by MrNaif2018** [comment]
@cmccandless I think now it is ready to merge, travis passes fine
